bot/connector-open-ai/src/main/kotlin/OpenAIConnector.kt

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (C) 2017/2025 SNCF Connect & Tech
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.tock.bot.connector.openai
+
+import ai.tock.bot.connector.ConnectorBase
+import ai.tock.bot.connector.ConnectorCallback
+import ai.tock.bot.connector.ConnectorData
+import ai.tock.bot.connector.ConnectorFeature.CAROUSEL
+import ai.tock.bot.connector.ConnectorType
+import ai.tock.bot.definition.IntentAware
+import ai.tock.bot.definition.StoryStepDef
+import ai.tock.bot.engine.BotRepository
+import ai.tock.bot.engine.ConnectorController
+import ai.tock.bot.engine.action.Action
+import ai.tock.bot.engine.action.ActionNotificationType
+import ai.tock.bot.engine.action.SendChoice
+import ai.tock.bot.engine.event.Event
+import ai.tock.bot.engine.event.MetadataEvent
+import ai.tock.bot.engine.user.PlayerId
+import ai.tock.bot.engine.user.PlayerType.bot
+import ai.tock.bot.engine.user.UserPreferences
+import ai.tock.shared.booleanProperty
+import ai.tock.shared.defaultLocale
+import ai.tock.shared.longProperty
+import ai.tock.shared.property
+import ai.tock.shared.security.auth.spi.WebSecurityHandler
+import ai.tock.shared.serialization.readJson
+import ai.tock.shared.serialization.writeJson
+import ai.tock.shared.vertx.WebSecurityCookiesHandler
+import ai.tock.shared.vertx.sendSsePing
+import ai.tock.shared.vertx.setupSSE
+import ai.tock.shared.vertx.vertx
+import com.aallam.openai.api.model.Model
+import com.aallam.openai.api.model.ModelId
+import io.vertx.core.http.HttpHeaders
+import io.vertx.core.http.HttpMethod.GET
+import io.vertx.core.http.HttpMethod.OPTIONS
+import io.vertx.core.http.HttpMethod.POST
+import io.vertx.ext.web.RoutingContext
+import io.vertx.ext.web.handler.CorsHandler
+import mu.KotlinLogging
+import java.time.Duration
+import java.util.Locale
+
+internal const val OPEN_AI_CONNECTOR_ID = "openai"
+
+/**
+ * The OpenAI connector type.
+ */
+val openAIConnectorType = ConnectorType(OPEN_AI_CONNECTOR_ID)
+
+private val corsPattern = property("tock_web_cors_pattern", ".*")
+private val sseEnabled = booleanProperty("tock_web_sse", false)
+
+class OpenAIConnector internal constructor(
+    val connectorId: String,
+    val path: String,
+    private val webSecurityHandler: WebSecurityHandler
+) : ConnectorBase(openAIConnectorType, setOf(CAROUSEL)) {
```

Comment:
Does this connector actually support carousels? It does not seem implemented here.

Reply:
fixed